.github/workflows/windows.yml

```diff
@@ -0,0 +1,93 @@
+name: mingw-w64
+
+on:
+  pull_request:
+  push:
+
+jobs:
+  mingw-w64:
+    name: Build and test
+    runs-on: windows-latest
+    steps:
+      - name: Checkout Git branch
+        uses: actions/checkout@v2
+
+      - name: Cache Msys2
+        id: cache-msys2
+        uses: actions/cache@v1.1.0
+        with:
+          path: msys64
+          key: ${{ runner.os }}-msys64
+
+      - name: Get Msys2
+        if: steps.cache-msys2.outputs.cache-hit != 'true'
+        run: |
+          Invoke-WebRequest -Outfile msys2.tar.xz -Uri "https://github.com/msys2/msys2-installer/releases/download/nightly-x86_64/msys2-base-x86_64-latest.tar.xz"
```

Comment:
The main reason why is because msys2 works on a rolling build so we could either get one from http://repo.msys2.org/distrib/ and update all of the packages, or use a `nightly` build that  I assume is built the same way as the regular distributions to reduce the amount we would have to download and update.

Plus, since it's from github's servers, I doubt that we would have issues downloading the tar.xz on the account that if github is down, their actions would be down also.

I'm not sure if there's a good way to reliably `pin` the version with pacman.

Perhaps the best way to make sure these are reproducible would be to upload the msys64 folder and reuse that forever without updating pacman (similar to what the cache job is doing right now).

Reply:
Good points, let's keep things this way then.